src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
-}
+  static removeDeadAnimals(animal) {
+    Animal.alive = Animal.alive.filter((deadAnimal) => {
+      return deadAnimal.name !== animal.name;
+    });
+  }
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 
+  thisClassName() {
+    return this.constructor.name;
+  }
+}
+Animal.alive = [];
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (!animal.hidden && !(animal.thisClassName() === 'Carnivore')) {
```

Comment:
Instead of using 'thisClassName' method to check if the animal is a carnivore, use 'instanceof' operator.